### PR TITLE
[NFC] Factor out and simplify minified name generation

### PR DESCRIPTION
**Pull request by tlively**
Simplifies the public API to not unnecessarily take an index and simplifies the
implementation to use a single integer as state rather than a vector of indices.

### CODE REVIEW

**Comment by tlively** [comment]
The lack of changes to test/passes/minify-imports-and-exports_all-features.wast shows that this is a non-functional change.

File: src/ir/names.h [hunk 1]
```diff
@@ -88,6 +87,23 @@ inline Name getValidElementSegmentName(Module& module, Name root) {
...
 
+class MinifiedNameGenerator {
+  // Reserved words we must not emit.
+  static std::unordered_set<std::string> reserved;
+
+  // Possible initial letters.
+  static std::string validInitialChars;
+
+  // Possible later letters.
+  static std::string validLaterChars;
```
Comment (kripken) [inline src/ir/names.h:98]: Do the statics need to be in the header? They could be only in the cpp I think, which seems better to me.

**Review by kripken — APPROVED** [review]
lgtm % comment

**Comment by tlively** [comment]
Not sure why macos is listed as "skipped" and alpine is taking so long as usual, but I'll land this manually because it otherwise looks good.

**Pull request merged**